Core/Src/internalResistance.c

```diff
@@ -0,0 +1,221 @@
+                }
+            }    
+        }
+    }
+    else
+    {
+        /*  If bad data was in the corresponding discrete voltage buffer, 
+            it is filled with a placeholder value to indicate it cannot be used for calculation */
+        currentAvgBuffer[avgBufferIndex] = -1000.0f;
+
+        // Static data good var is reset for the next discrete buffer filling cycle
+        currentDataGood = true;
+    }
+
+    // Cycle the circular average buffers and loop if necessary
+    if (++avgBufferIndex >= AVERAGE_BUFFER_SIZE)
+    {
+        avgBufferIndex = 0;
+    }
+
+    // Max and min data initilized to far left and right values
+    float maxCurrent = -1000.0f;
+    float minCurrent = 1000.0f;
+
+    // Default max and min index is 0
+    uint32_t maxCurrentIndex = 0;
+    uint32_t minCurrentIndex = 0;
+
+    // Calculate max and min currents in buffer
+    for(int32_t i = 0; i < AVERAGE_BUFFER_SIZE; i++)
+    {
+        // If average current buffer containes placeholder value, -1000.0f, it is excluded from the max/min calc
+        if(currentAvgBuffer[i] > -900.0f)
+        {
+            if(currentAvgBuffer[i] > maxCurrent)
+            {
+                maxCurrentIndex = i;
+            }
+            else if(currentAvgBuffer[i] < minCurrent)
+            {
+                minCurrentIndex = i;
+            }
+        }
+    }
+
+    //  Calculate greatest dI and dV contained in buffer
+    /*  If avg buffer saturated with -1000.0f placeholder values, default index 0 will still be set, 
+        and delta current will equal 0 (-1000.0f - -1000.0f)    */
+    float deltaCurrent = currentAvgBuffer[maxCurrentIndex] - currentAvgBuffer[minCurrentIndex];
+
+    // If delta current threshold is met, internal resistance can be calulated and updated
+    // This threshold should always exceed 0, so if avg buffer is saturated with -1000.0f, operation will be skipped
+    if(fabs(deltaCurrent) >= IR_CALC_MIN_CURRENT_DELTA)
+    {
+        // Cycle through every BMB and every brick in the accumulator
+        for(int32_t i = 0; i < NUM_BMBS_PER_PACK; i++)
+        {
+            for(int32_t j = 0; j < NUM_BRICKS_PER_BMB; j++)
+            {
+                // If average current buffer containes placeholder value, -1000.0f, it is excluded from this calculation cycle
+                if((voltageAvgBuffer[i][j][maxCurrentIndex] > -900.0f) && (voltageAvgBuffer[i][j][minCurrentIndex] > -900.0f))
+                {
+                    float deltaVoltage = voltageAvgBuffer[i][j][maxCurrentIndex] - voltageAvgBuffer[i][j][minCurrentIndex];
+                    bms->bmb[i].brickResistance[j] = deltaVoltage / deltaCurrent;
+                }
+            }    
+        }
+    }
+
+    // Static data ready vars set to 0 for next discrete buffer filling cycle 
+    currentDataReady = false;
+    voltageDataReady = false;
+
+    // Reset discrete buffer pointers so data begins filling from index 0 
+    discreteCurrentBufferIndex = 0;
```

Comment:
I think this should be moved to the other function which handles the average buffer updating